docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,278 @@
+## Documentation for how the dataset was made
+
+0. We used the environment defined in `make_peptides_ml_dataset.yml` to make the data set below.
+1. Download the peptipedia database:  https://drive.google.com/file/d/1x3yHNl8k5teHlBI2FMgl966o51s0T8i_/view?usp=sharing. The link for this database is provided by the [peptipedia GitHub repo](https://github.com/ProteinEngineering-PESB2/Peptipedia2.0).
+2. Unzip the database.
+3. The unzipped archive contains three files, `backup_sql.zip`, `peptipedia_csv.zip`, and `peptipedia_fasta.zip`.
+   Unzipping `peptipedia_fasta.zip` results in a FASTA file `peptipedia.fasta` that contains all of the peptide sequences from the peptipedia database.
+   Unzipping `peptipedia_csv.zip` results in a CSV file `peptipedia.csv` that records the peptide id, peptide sequence, and predicted bioactivity labels.
+   Unzipping `backup_sql.zip` results in a postgres SQL dump file `backup_dump.sql`.
+   This file contains metadata associated with the peptide sequences.
+   To extract the metadata to separate CSV tables, we used the python code:
+   ```py
+   import re
+   
+   output_file_prefix = 'table_'
+   current_table = None
+   output_file = None
+   
+   with open('backup_dump.sql', 'r') as dump_file:
+       for line in dump_file:
+           # Check for COPY command start
+           if line.startswith('COPY'):
+               table_name = line.split()[1]  # Assumes table name is the second word
+               current_table = table_name
+               output_file = open(f'{output_file_prefix}{current_table}.csv', 'w')
+           elif line.startswith('\.'):
+               # End of data for the current table
+               if output_file:
+                   output_file.close()
+                   output_file = None
+           elif output_file:
+               # Convert tab-separated values to comma-separated values
+               csv_line = line.replace('\t', ',')
+               output_file.write(csv_line)
```

Comment:
can you note what files are expected to be extracted e.g. `table_public.gene_ontology` etc?

Reply:
added! I'll also add the extracted CSVs to wherever I upload the database as well.